templates/kobotoolbox/0/docker-compose.yml

```diff
@@ -207,88 +207,126 @@ services:
+      ENKETO_URL: ${KOBOTOOLBOX_PROTOCOL}://${ENKETO_SERVER_NAME}
+      CSRF_COOKIE_DOMAIN: ${CSRF_COOKIE_DOMAIN}
+      DJANGO_ALLOWED_HOSTS: ${DJANGO_ALLOWED_HOSTS}
+      KOBOFORM_SERVER_NAME: ${KOBOFORM_SERVER_NAME}
+      KOBOCAT_SERVER_NAME: ${KOBOCAT_SERVER_NAME}
+      ENKETO_SERVER_NAME: ${ENKETO_SERVER_NAME}
 
       {{- end }}
 
-      - KOBOCAT_DJANGO_DEBUG=False
-      - TEMPLATE_DEBUG=False
-      - DJANGO_SETTINGS_MODULE=onadata.settings.kc_environ
-      - ENKETO_VERSION=Express
-      - DATABASE_URL=postgis://kobo:kobo@postgres:5432/kobotoolbox
-      - KOBO_POSTGRES_DB_NAME=kobotoolbox
-      - KOBO_POSTGRES_USER=kobo
-      - KOBOCAT_BROKER_URL=amqp://kobocat:kobocat@rabbit:5672/kobocat
+      KOBOCAT_DJANGO_DEBUG: False
+      TEMPLATE_DEBUG: False
+      DJANGO_SETTINGS_MODULE: onadata.settings.kc_environ
+      ENKETO_VERSION: Express
+      DATABASE_URL: postgis://kobo:kobo@postgres:5432/kobotoolbox
+      KOBO_POSTGRES_DB_NAME: kobotoolbox
+      KOBO_POSTGRES_USER: kobo
+      KOBOCAT_BROKER_URL: amqp://kobocat:kobocat@rabbit:5672/kobocat
 
-      - KPI_WEB_SERVER=${PROXY_WEB_SERVER}
-      - KOBOCAT_WEB_SERVER=${PROXY_WEB_SERVER}
-      - KOBOCAT_STATIC_FILES_SERVER=Nginx
-      - KPI_STATIC_FILES_SERVER=Nginx
+      KPI_WEB_SERVER: ${PROXY_WEB_SERVER}
+      KOBOCAT_WEB_SERVER: ${PROXY_WEB_SERVER}
+      KOBOCAT_STATIC_FILES_SERVER: Nginx
+      KPI_STATIC_FILES_SERVER: Nginx
 
-      - ENKETO_PROTOCOL=${KOBOTOOLBOX_PROTOCOL}
+      ENKETO_PROTOCOL: ${KOBOTOOLBOX_PROTOCOL}
 
-      - EMAIL_BACKEND=django.core.mail.backends.smtp.EmailBackend
+      EMAIL_BACKEND: django.core.mail.backends.smtp.EmailBackend
 
       {{- if eq .Values.USE_EXTERNAL_SMTP "true" }}
 
-      - EMAIL_HOST=${EMAIL_HOST}
-      - EMAIL_PORT=${EMAIL_PORT}
-      - EMAIL_HOST_USER=${EMAIL_HOST_USER}
-      - EMAIL_HOST_PASSWORD=${EMAIL_HOST_PASSWORD}
-      - EMAIL_USE_TLS=${EMAIL_USE_TLS}
+      EMAIL_HOST: ${EMAIL_HOST}
+      EMAIL_PORT: ${EMAIL_PORT}
+      EMAIL_HOST_USER: ${EMAIL_HOST_USER}
+      EMAIL_HOST_PASSWORD: ${EMAIL_HOST_PASSWORD}
+      EMAIL_USE_TLS: ${EMAIL_USE_TLS}
 
       {{- else }}
 
-      - EMAIL_HOST=smtp
-      - EMAIL_PORT=25
-      - EMAIL_HOST_USER=noreply
-      - EMAIL_HOST_PASSWORD=docker
-      - EMAIL_USE_TLS=False
+      EMAIL_HOST: smtp
+      EMAIL_PORT: 25
+      EMAIL_HOST_USER: noreply
+      EMAIL_HOST_PASSWORD: docker
+      EMAIL_USE_TLS: False
 
       {{- end }}
-      - DEFAULT_FROM_EMAIL=${KOBO_SUPPORT_EMAIL}
 
-      - CELERYD_TASK_TIME_LIMIT=${TASK_TIMEOUT}
-      - CELERYD_TASK_SOFT_TIME_LIMIT=${TASK_SOFT_TIMEOUT}
+      {{- if eq .Values.USE_KOBOCAT_THEME_APP "True" }}
+
+      DJANGO_SETTINGS_MODULE: "custom_kobocat_theme.settings"
+      PYTHONPATH: "/srv/src/custom_kobocat_theme"
+      CUSTOM_KOBOCAT_THEME_APP_PIP_URL: ${CUSTOM_KOBOCAT_THEME_APP_PIP_URL}
+      CUSTOM_KOBOCAT_THEME_DEPLOY_KEY:
+      |-
+         -----BEGIN OPENSSH PRIVATE KEY-----
```

Comment:
Ahahaha. Do not put private key here @boney-bun . This is a public repo, anyone can see it.
Please change your private key as this is already leaked.

Create a rancher question to ask the user private key, and experiment if that method is working (ask user the private key and put the variable here).

You could do something like:
CUSTOM_KOBOCAT_THEME_DEPLOY_KEY: 
|-
${CUSTOM_KOBOCAT_THEME_DEPLOY_KEY}